internal/controllers/pod_controller.go

```diff
@@ -164,11 +164,31 @@ func (r *PodReconciler) labelPod(ctx context.Context, pod corev1.Pod, result val
 	return nil
 }
 
-// TODO: This function should check images not the whole spec.
-func (r *PodReconciler) areImagesChanged(oldObject runtime.Object, newObject runtime.Object) bool {
-	oldPod := oldObject.(*corev1.Pod)
-	newPod := newObject.(*corev1.Pod)
-	return !reflect.DeepEqual(oldPod.Spec.InitContainers, newPod.Spec.InitContainers) || !reflect.DeepEqual(oldPod.Spec.Containers, newPod.Spec.Containers)
+func areImagesChanged(oldObject runtime.Object, newObject runtime.Object) bool {
+	oldImages := getPodImages(oldObject.(*corev1.Pod))
+	newImages := getPodImages(newObject.(*corev1.Pod))
+	if len(oldImages) != len(newImages) {
+		return true
+	}
+	sort.Strings(oldImages)
+	sort.Strings(newImages)
```

Comment:
What if we have additional image in new pod?

Reply:
IMO all is ok. Maybe I don't understand this question.